src/app/reporting/Engine.cpp

```diff
@@ -79,10 +79,10 @@ std::optional<CHIP_ERROR> ValidateReadAttributeACL(DataModel::Provider * dataMod
     // privilege and default to kView (this is correct for global attributes and a reasonable check
     // for others)
     Privilege requiredPrivilege = Privilege::kView;
-    if (info.has_value() && info->readPrivilege.has_value())
+    if (info.has_value())
     {
-        // attribute exists and is readable, set the correct read privilege
-        requiredPrivilege = *info->readPrivilege;
+        // attribute exists, and if is readable, set the correct read privilege
+        requiredPrivilege = info->GetReadPrivilege().has_value() ? info->GetReadPrivilege().value() : requiredPrivilege;
```

Comment:
assign to a temporary variable here, so it is clear that we call has_value and value on the same const object.

Otherwise the compiler can say `GetReadPrivilege may return a different value on each call, so this is a potential bug` 

Applies throughout 

```suggestion
        const readPrivilege = info->GetReadPrivilege();
        requiredPrivilege = readPrivilege.has_value() ? *readPrivilege: requiredPrivilege;
```

Reply:
maybe this is even more readable:

```suggestion
        requiredPrivilege = info->GetReadPrivilege().value_or(requiredPrivilege);
```

Reply:
Agree the second one would be clearer, hopefully it works on apple builds as I remember the optional interface not quite complete in xcode's clang until later versions

Reply:
I tried this solution, but clang-tidy didn't like the temporary variable assignment:

```
const readPrivilege = info->GetReadPrivilege();
                      ^
error: unchecked access to optional value [bugprone-unchecked-optional-access,-warnings-as-errors]
```

Reply:
Using `.value_or()` fixed the `clang-tidy` error:

```
    Privilege requiredPrivilege = Privilege::kView;
    if (info.has_value())
    {
        // if attribute exists and is readable, set the correct read privilege; otherwise, set default value
        requiredPrivilege = info->GetReadPrivilege().value_or(requiredPrivilege);
    }
```